### ISSUE TITLE
`arr.min()/max()` panicks when one of the values if `None` and the number of rows is larger than 2

### ISSUE THREAD

**Issue by etiennebacher** [pola-rs/polars#14359]
### Checks

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

```python
import os
os.environ['POLARS_VERBOSE']='1'
import polars as pl

### Works (2 rows, 1 None value)
pl.DataFrame(
    data={"a": [[1, 2], [3, None]]},
    schema={"a": pl.Array(pl.Int64, 2)},
).select(pl.col("a").arr.max())

### Works (3 rows, 0 None value)
pl.DataFrame(
    data={"a": [[1, 2], [3, 4], [5, 6]]},
    schema={"a": pl.Array(pl.Int64, 2)},
).select(pl.col("a").arr.max())

### Panicks (3 rows, 1 None value)
pl.DataFrame(
    data={"a": [[1, 2], [3, 4], [None, 5]]},
    schema={"a": pl.Array(pl.Int64, 2)},
).select(pl.col("a").arr.max())
```

### Log output

```shell
thread '<unnamed>' panicked at crates\polars-arrow\src\bitmap\immutable.rs:156:24:
range end index 2 out of range for slice of length 1
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Traceback (most recent call last):
  File "<stdin>", line 4, in <module>
  File "C:\Python311\Lib\site-packages\polars\dataframe\frame.py", line 8142, in select
    return self.lazy().select(*exprs, **named_exprs).collect(_eager=True)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python311\Lib\site-packages\polars\lazyframe\frame.py", line 1940, in collect
    return wrap_df(ldf.collect())
                   ^^^^^^^^^^^^^
pyo3_runtime.PanicException: range end index 2 out of range for slice of length 1
```

### Issue description

`arr.max()` and `arr.min()` panick if there are more than 2 rows in the data and if one the array values is `None`.

### Expected behavior

```
┌─────┐
│ a   │
│ --- │
│ i64 │
╞═════╡
│ 2   │
│ 4   │
│ 5   │
└─────┘
```

### Installed versions

<details>

```
--------Version info---------
Polars:               0.20.7
Index type:           UInt32
Platform:             Windows-10-10.0.19044-SP0
Python:               3.11.0 (main, Oct 24 2022, 18:26:48) [MSC v.1933 64 bit (AMD64)]

----Optional dependencies----
adbc_driver_manager:  <not installed>
cloudpickle:          <not installed>
connectorx:           <not installed>
deltalake:            <not installed>
fsspec:               2023.6.0
gevent:               <not installed>
hvplot:               <not installed>
matplotlib:           3.7.1
numpy:                1.24.3
openpyxl:             <not installed>
pandas:               2.0.3
pyarrow:              12.0.1
pydantic:             <not installed>
pyiceberg:            <not installed>
pyxlsb:               <not installed>
sqlalchemy:           <not installed>
xlsx2csv:             <not installed>
xlsxwriter:           <not installed>
```

</details>

**Comment by deanm0000** [comment]
@reswqa I think you recently fixed one like this.

**Comment by eitsupi** [comment]
ref: pola-rs/r-polars#790

It seems this is a bug of rustc will be fixed in 1.77.0 (rust-lang/rust#119352)

**Comment by etiennebacher** [comment]
This is solved as of 0.20.21 (probably in #15654)
